### PR TITLE
refactor: Update editor to use more C++11-isms

### PR DESCRIPTION
**Pull request by Teklad**
This just updates some of the older code in editor to use C++11 constructs.  Also fixes a small regression from the LanguageService PR where frmpreferences doesn't select the correct highlighting mode for editor preview.

### CODE REVIEW

**Comment by Teklad** [comment]
I've made a few extra changes since this PR was started.  If there's any you don't like I can revert each of them individually.  This was done intentionally so we can get the most out of this PR.

I'll continue to experiment with refactoring and make small changes here and there for a bit.

File: src/ui/EditorNS/editor.cpp [hunk 1]
```diff
@@ -141,37 +141,33 @@ namespace EditorNS
...
             unsigned int id = rgx.capturedTexts()[1].toInt();
 
             // Look into the list of callbacks
-            for (auto it = this->asyncReplies.begin(); it != this->asyncReplies.end(); ++it) {
-                if (it->id == id) {
-                    auto cb = it->callback;
-                    it->value->set_value(data);
-                    this->asyncReplies.erase(it);
-
-                    if (cb != 0) {
```
Comment (JuBan1) [inline src/ui/EditorNS/editor.cpp]: Change you change that to `cb != nullptr`? It quenches clang warnings about using zero as nullpointer constant.

File: src/ui/EditorNS/editor.cpp [hunk 2]
```diff
@@ -141,37 +141,33 @@ namespace EditorNS
...
+            auto it = std::find_if(asyncReplies.begin(), asyncReplies.end(), [&id] (const AsyncReply& reply) {
+                return (reply.id == id);
+            });
+            if (it == asyncReplies.end())
+                return;
+            auto cb = it->callback;
+            it->value->set_value(data);
+            this->asyncReplies.erase(it);
 
+            if (cb != 0) {
```
Comment (JuBan1) [inline src/ui/EditorNS/editor.cpp]: Change you change that to `cb != nullptr`? It quenches clang warnings about using zero as nullpointer constant.

**Comment by danieleds** [comment]
We have to check if it compiles on Ubuntu 14.04. I've had a few bad experiences in the past where I introduced some nice new syntax and everything failed compiling when I was shipping the deb.

**Comment by danieleds** [comment]
Also because of that, can you fix the regression with another PR?

**Comment by Teklad** [comment]
So is Ubuntu 14.04 the bare minimum we'll support?  I'd figure we'd go for 16.04 considering the looming WebEngine conversion and other what nots.

Also, sure thing... let me cherry-pick that out of there real quick.

**Comment by danieleds** [comment]
Unfortunately, yes. Ubuntu 14.04 is still alive and a lot of users (you'd be surprised) are using it.

I still need to prepare something real quick to show you the stats we collect.

We'll stop sending updates after webengine2 is ready, but for now we have this weight

**Comment by Teklad** [comment]
I look forward to that day.  We've got a lot of spaghetti thrown around everywhere that could be greatly improved using proper C++11 syntax.

File: src/ui/EditorNS/editor.cpp [hunk 3]
```diff
@@ -141,37 +139,33 @@ namespace EditorNS
         if (msg.startsWith("[ASYNC_REPLY]")) {
             QRegExp rgx("\\[ID=(\\d+)\\]$");
 
-            if(rgx.indexIn(msg) == -1)
-                return;
-
-            if (rgx.captureCount() != 1)
+            if (rgx.indexIn(msg) == -1 || rgx.captureCount() != 1)
```
Comment (danieleds) [inline src/ui/EditorNS/editor.cpp]: Messing with on_proxyMessageReceived is going to create a lot of merge conflicts in the webengine2 and webengine2-promise branches I'm working on... It's getting hard to fix conflicts at every PR xD.

I think it's not worth the work, especially because this code is going to disappear anyway. Could you revert these changes, or at least do them on those branches?

**Comment by Teklad** [comment]
Done.  Even reverts tend to get a little hairy after enough commits have been made, lol.

**Comment by danieleds** [comment]
In Editor::print(), can you correct the theme name from "Default" to "default"?

**Comment by Teklad** [comment]
Done.
